src/main/java/hudson/plugins/promoted_builds/parameters/PromotedBuildParameterDefinition.java

```diff
@@ -160,33 +206,53 @@ public String getHelpFile() {
         public ParameterDefinition newInstance(StaplerRequest req, JSONObject formData) throws FormException {
             return req.bindJSON(PromotedBuildParameterDefinition.class, formData);
         }
-
+        
         /**
          * Checks the job name.
          */
         public FormValidation doCheckJobName(@AncestorInPath Item project, @QueryParameter String value ) {
+            final Jenkins jenkins = Jenkins.getInstance();
+            if (jenkins == null) {
+                return FormValidation.error("Jenkins instance is not ready");
+            }
             if (!project.hasPermission(Item.CONFIGURE) && project.hasPermission(Item.EXTENDED_READ)) {
                 return FormValidation.ok();
             }
+            
             project.checkPermission(Item.CONFIGURE);
```

Comment:
What should it return if no permissions?
Maybe wrap everything in check with try and catch error that will return FalidationForm.error(ex.getMessage())? And then throw illegalStateException for null jenkins?